pkg/upgrade/control_plane.go

```diff
@@ -815,3 +1015,121 @@ func (u *ControlPlaneUpgrader) UpdateProviderIDsToNodes() error {
 
 	return nil
 }
+
+func (u *ControlPlaneUpgrader) waitForProviderID(ns, name string, timeout time.Duration) (string, error) {
+	u.log.Info("waitForMachineProviderID start", "namespace", ns, "name", name)
+	var providerID string
+	err := wait.PollImmediate(5*time.Second, timeout, func() (bool, error) {
+		machine := &clusterv1.Machine{}
+		if err := u.managementClusterClient.Get(context.TODO(), ctrlclient.ObjectKey{Name: name, Namespace: ns}, machine); err != nil {
+			return false, errors.WithStack(err)
```

Comment:
Is this related to https://github.com/vmware/cluster-api-upgrade-tool/issues/96 ?

Reply:
+1 to logging the error though to get to the full timeout.

Reply:
It's possible it's related. I saw issues with the capd/nginx load balancer where every 4th request would fail (3 original machines + 1 new machine that hadn't finished bootstrapping).

Reply:
seems like there isn't much we can do about it from this side (upgrade tool) except your suggested fix